### ISSUE TITLE
[BUG]Error UnsupportedOperationException on Delta Table with Azure Databricks

### ISSUE THREAD

**Issue by chrisVietDev** [NVIDIA/spark-rapids#3903]
**Describe the bug**
Got "UnsupportedOperationException" for reading DeltaLake Table

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

```
spark.conf.set('spark.rapids.sql.enabled','true')
spark.conf.set('spark.rapids.sql.explain', 'ALL')
spark.conf.set('spark.rapids.sql.incompatibleOps.enabled', 'true')
spark.conf.set('spark.rapids.sql.batchSizeBytes', '512M')
spark.conf.set('spark.rapids.sql.reader.batchSizeBytes', '768M')

root_path = "dbfs:/mnt/overwatch" 

# ERROR Scenario 1: java.lang.UnsupportedOperationException: Converting to GPU for 
dfCluster = spark.read.format("delta").load(f"{root_path}/global_share/cluster_gold")

dfCluster.count()
```

Error message

```
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 4.0 failed 4 times, most recent failure: Lost task 0.3 in stage 4.0 (TID 16) (10.241.16.91 executor 0): java.lang.UnsupportedOperationException: Converting to GPU for StructType(StructField(path,StringType,true), StructField(partitionValues,MapType(StringType,StringType,true),true), StructField(size,LongType,false), StructField(modificationTime,LongType,false), StructField(tags,MapType(StringType,StringType,true),true), StructField(partitionValues_parsed,StructType(StructField(organization_id,StringType,true),
...

is not currently supported
```

If I save the data to the Parquet it work fine with GPU

**Expected behavior**
This query should work fine with GPU

**Environment details (please complete the following information)**
 Databricks (ML 8.2 GPU)

```
sudo wget -O /databricks/jars/rapids-4-spark_2.12-21.08.0.jar https://repo1.maven.org/maven2/com/nvidia/rapids-4-spark_2.12/21.08.0/rapids-4-spark_2.12-21.08.0.jar
```

**Additional context**

**Comment by viadea** [comment]
As per my test using 21.12 snapshot jars which have the fix from https://github.com/NVIDIA/spark-rapids/issues/1304, this issue is resolved.

However the issue-3872 still exists.

**Comment by jlowe** [comment]
> As per my test using 21.12 snapshot jars which have the fix from #1304, this issue is resolved.

Thanks, @viadea.  Closing this as a duplicate of #1304 which is now fixed.
